### DISCUSSION TITLE
System.Text.Json: Deserializing enums with spaces in their names

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#47441] [Q&A]
With Newtonsoft you could use ```EnumDataMember``` to allow deserializing an enum that has a space in the name.

How do you do that with STJ?  I don't really need a whole converter for that do I?

**Comment by huoyaoyuan** [comment]
Try `[JsonPropertyName]`?
`JsonNamingPolicy` works for enum names, but it is specified at top level.

**Comment by TonyValenti** [comment]
@huoyaoyuan I just tested this and it doesn't seem to compile.  What am I doing wrong?

```
using System;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace ConsoleApp33 {
    class Program {
        static void Main(string[] args) {

            var V = new Field() {
                FieldName = FieldName.LastName
            };

            var Options = new JsonSerializerOptions() {
                Converters = {
                    new JsonStringEnumConverter(),
                }
            };

            var Text = JsonSerializer.Serialize(V, Options);
            var Text2 = @"{""FieldName"":""Last Name""}";
            var V2 = JsonSerializer.Deserialize<Field>(Text2, Options);

        }


    }

    public class Field {
        public FieldName FieldName { get; set; }
    }

    public enum FieldName {
        [JsonPropertyName("First Name")]
        FirstName,

        [JsonPropertyName("Last Name")]
        LastName,

        [JsonPropertyName("0xA3C")]
        InternalReference,

    }

}

```

**Reply by huoyaoyuan** [reply]
`JsonPropertyName` is applicable to enum members. I have no idea why you don't get it compiled.

**Reply by TonyValenti** [reply]
I just tested on .NET 5.0 and it compiles but it doesn't respect the value on serialization and it crashes on deserialization.

**Reply by huoyaoyuan** [reply]
For 5.0, it should mean that you have no way other than custom `JsonNamingPolicy` or `JsonConverter`.
But since the attribute is applicable, you may propose a feature request to make it work in 6.0.
